spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryUtils.java

```diff
@@ -591,9 +617,14 @@ public static String createCountQueryFor(String originalQuery, @Nullable String
 
 			String replacement = useVariable ? SIMPLE_COUNT_VALUE : complexCountValue;
 
-			String alias = QueryUtils.detectAlias(originalQuery);
-			if ("*".equals(variable) && alias != null) {
-				replacement = alias;
+			if (nativeQuery && (variable.contains(",") || "*".equals(variable))) {
```

Comment:
What if it is a native query and the variable is null 😬

Reply:
If you run into a bug or you can reproduce such a case, please file a bug report.

Reply:
already present https://github.com/spring-projects/spring-data-jpa/issues/2812